lib/src/presentation/widget/list_of_cards.dart

```diff
@@ -34,55 +34,58 @@ class ListOfCards extends StatelessWidget {
-            }
-            snapshot.data?.listOfCards!.forEach((hearthstoneCard) {
-              children.add(
-                GestureDetector(
-                  onTap: () {
-                    Navigator.push(
-                      context,
-                      MaterialPageRoute(
-                        builder: (context) => CardView(
-                          card: hearthstoneCard,
-                        ),
+            switch (snapshot.data!.state) {
+              case ApiResponseState.empty:
+                return const NotFound();
+              case ApiResponseState.success:
+                snapshot.data?.listOfCards!.forEach((hearthstoneCard) {
+                  children.add(
+                    GestureDetector(
+                      onTap: () {
+                        Navigator.push(
+                          context,
+                          MaterialPageRoute(
+                            builder: (context) => CardView(
+                              card: hearthstoneCard,
+                            ),
+                          ),
+                        );
+                      },
+                      child: GridCardItem(
+                        image: hearthstoneCard.image,
+                        name: hearthstoneCard.name ?? Strings.emptyString,
+                        cardSet: hearthstoneCard.cardSet ?? Strings.emptyString,
+                        type: hearthstoneCard.type ?? Strings.emptyString,
                       ),
-                    );
-                  },
-                  child: GridCardItem(
-                    image: hearthstoneCard.image,
-                    name: hearthstoneCard.name ?? Strings.emptyString,
-                    cardSet: hearthstoneCard.cardSet ?? Strings.emptyString,
-                    type: hearthstoneCard.type ?? Strings.emptyString,
-                  ),
-                ),
-              );
-            });
-          } else if (snapshot.hasError) {
-            children.add(
-              Text(
-                snapshot.error.toString(),
-              ),
-            );
+                    ),
+                  );
+                });
+                return GridView.count(
+                  padding: Styles.smallPadding,
+                  crossAxisCount: Styles.gridItemsPerRow,
+                  mainAxisSpacing: Styles.gritItemSpacing,
+                  crossAxisSpacing: Styles.gritItemSpacing,
+                  childAspectRatio: Styles.gridItemAspectRatio,
+                  children: children,
+                );
+              case ApiResponseState.loading:
+                return const Center(
+                  child: CircularProgressIndicator(),
+                );
+              case ApiResponseState.error:
+                return Text(
+                  snapshot.error.toString(),
+                );
+              default:
+                return const Center(
+                  child: Home(),
+                );
```

Comment:
This always ocurres the first time that you open the app right?

Reply:
Yes, I needed a default case for the switch to return something, so I choose to use the initial state, but it's not explicit. The same screen is shown if the stream has no data.